### PR TITLE
Space Travelers Project

### PR DESCRIPTION
**Pull request by Haadiiii**
## Microverse Space Travelers Hub Project ⏬  👍 

in this feature:

> Rocket 💯 

- setup the rockets component
- -setup the action
- Created unit test
- setup the reducers
- setup the reducers
- add the styling

> missions 💯 

- setup the mission component
- -setup the action
- Created unit test
- setup the reducers
- add the styling

> Profile 💯 

- setup the Profile component
- -setup the action
- Created unit test
- setup the reducers
- add the styling

**Fixed  linters Errors** 💯 

image

### CODE REVIEW

File: src/Redux/mission/Mission.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      return {
+        id: key,
+        ...mission,
+      };
+    });
+    dispatch({
+      type: SHOW_MISSION,
+      payload: missions,
+    });
+    return missions;
```
Comment (DammyShittu) [inline src/Redux/mission/Mission.js]: - [ ] Only the needed data from the missions should be dispatched. Now you are dispatching unneeded data to the store. From the requirements, only these data should be saved: `mission_id`, `mission_name`, `description`. The rest should be discarded.

  midd

File: src/Redux/mission/Mission.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      return {
+        ...state,
+        missionArray: action.payload,
+      };
+    case SHOW_STATUS:
+      return {
+        missionArray: state.missionArray.map((mission) => (mission.mission_id === action.payload
+          ? { ...mission, reserved: true }
+          : mission)),
+      };
```
Comment (DammyShittu) [inline src/Redux/mission/Mission.js]: - [ ] You seem to have implemented your reservation status wrongly. As can see a reserved mission should have the `reserved` attribute set to `true`, but in your case, it is set to `false`, and vice versa. Kindly rectify that.

  ddp

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @killy10o10 @Haadiiii,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/mission/Mission.js (hunk 1 above)
Comment (Haadiiii) [reply src/Redux/mission/Mission.js:26]: hey @DammyShittu , thanks for the review and i did what you suggested 👍

File: src/Redux/mission/Mission.js (hunk 2 above)
Comment (Haadiiii) [reply src/Redux/mission/Mission.js:53]: hey @DammyShittu ,
thanks for this, but I think you console log to the wrong destination see the snaps below.. but yeah I made some changes.
## 1 see the mission below ⏬ 
image

## 2 see the profile below.  ⏬ 

new1

**Review by roshan-bajgain — APPROVED** [review]
Hi @Haadiiii and @killy10o10, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**